api/tests/Feature/PoolApplicationTest.php

```diff
@@ -15,100 +16,184 @@
 use Database\Seeders\SkillFamilySeeder;
 use Database\Seeders\SkillSeeder;
 use Carbon\Carbon;
+use Illuminate\Foundation\Testing\WithFaker;
 
 class PoolApplicationTest extends TestCase
 {
     use RefreshDatabase;
     use MakesGraphQLRequests;
     use RefreshesSchemaCache;
+    use WithFaker;
+
+    protected $guestUser;
+    protected $applicantUser;
+    protected $teamUser;
+    protected $adminUser;
+    protected $uuid;
+    protected $toBeDeleted;
+    protected $teamName = "application-test-team";
+
+    protected $queryDocument =
+    /** @lang GraphQL */
+    '
+        query poolCandidate($id: UUID!) {
+            poolCandidate(id: $id) {
+                status
+            }
+        }
+    ';
+
+    protected $createMutationDocument =
+    /** @lang GraphQL */
+    '
+        mutation createApplication($userId: ID!, $poolId: ID!){
+            createApplication(userId: $userId, poolId: $poolId) {
+                user {
+                    id
+                }
+                pool {
+                    id
+                }
+                status
+            }
+        }
+    ';
+
+    protected $archiveMutationDocument =
+    /** @lang GraphQL */
+    '
+        mutation archiveApplication($id: ID!) {
+            archiveApplication(id: $id) {
+                archivedAt
+            }
+        }
+    ';
+
+    protected $deleteMutationDocument =
+    /** @lang GraphQL */
+    '
+        mutation deleteApplication($id: ID!) {
+            deleteApplication(id: $id)
+        }
+    ';
+
+    protected $submitMutationDocument =
+    /** @lang GraphQL */
+    '
+        mutation submitTest($id: ID!, $sig: String!) {
+            submitApplication(id: $id, signature: $sig) {
+                submittedAt
+                signature
+                status
+            }
+        }
+    ';
 
     protected function setUp(): void
     {
         parent::setUp();
+
+        // Run necessary seeders
+        $this->seed(ClassificationSeeder::class);
+        $this->seed(GenericJobTitleSeeder::class);
+        $this->seed(SkillFamilySeeder::class);
+        $this->seed(SkillSeeder::class);
+        $this->seed(RolePermissionSeeder::class);
+
+        $this->setUpFaker();
         $this->bootRefreshesSchemaCache();
+
+        Team::factory()->create([
+            'name' => $this->teamName,
+        ]);
+
+        $this->guestUser = User::factory()->create([
+            'email' => 'guest-user@test.com',
+            'sub' => 'guest-user@test.com',
+        ]);
+        $this->guestUser->syncRoles([
+            "guest",
+        ]);
```

Comment:
This was required to essentially test unauthenticated users since `actingAs` requires a user be passed in and we currently have a user set by default for PHPUnit.

Reply:
Hmm. Awkward. 🤔

Reply:
Well, this is fine for now. Better solution can wait for another issue: #5859.